### PR TITLE
enh(ci): package centreon-awie using nfpm

### PR DESCRIPTION
**Pull request by hamzabessa**
## Description

package centreon-awie using nfpm

**Fixes** MON-24164

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon-awie/packaging/centreon-awie.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    dst: "/usr/share/centreon/www/modules/centreon-awie"
+    file_info:
+      mode: 0664
+
+overrides:
+  deb:
+    depends:
+      - debhelper-compat (=12)
+      - lsb-base
+
```
Comment (kduret) [inline centreon-awie/packaging/centreon-awie.yaml:28]:
```suggestion
```

**Review by kduret — COMMENTED** [review]
you must delete old rpm and deb packaging files

File: .github/workflows/awie.yml [hunk 2]
```diff
@@ -60,21 +60,56 @@ jobs:
...
+            -exec grep -qE '(@CENTREON_ETC@)|(@BIN_DIR@)|(CENTREON_DIR)|(CENTREON_LOG)|(CENTREON_VARLIB)|(MODULE_NAME)|(DB_CENTSTORAGE)' {} ';' \
+            -exec sed -i -e "s|@CENTREON_ETC@|${centreon_etc}|g" \
+                          -e "s|@BIN_DIR@|/usr/bin|g" \
+                          -e "s|@CENTREON_DIR@|${centreon_dir}|g" \
+                          -e "s|@CENTREON_LOG@|${centreon_log}|g" \
+                          -e "s|@CENTREON_VARLIB@|${centreon_varlib}|g" \
+                          -e "s|@MODULE_NAME@|centreon-awie|g" \
+                          -e "s|@DB_CENTSTORAGE@|centreon_storage|g" \
+                      {} ';'
+        shell: bash
```
Comment (kduret) [inline .github/workflows/awie.yml:89]: I have check the source code of awie and these macros do not exist
```suggestion
```

File: .github/workflows/awie.yml [hunk 3]
```diff
@@ -60,21 +60,56 @@ jobs:
...
+        with:
+          nfpm_file_pattern: "centreon-awie/packaging/*.yaml"
+          distrib: ${{ matrix.distrib }}
+          package_extension: ${{ matrix.package_extension }}
+          major_version: ${{ needs.get-version.outputs.major_version }}
+          minor_version: ${{ needs.get-version.outputs.minor_version }}
+          release: ${{ needs.get-version.outputs.release }}
+          arch: all
+          commit_hash: ${{ github.sha }}
+          cache_key: cache-${{ github.sha }}-${{ matrix.package_extension}}-centreon-awie-${{ matrix.distrib }}-${{ github.head_ref || github.ref_name }}
```
Comment (kduret) [inline .github/workflows/awie.yml:102]: the cache key must be the same than the deliver jobs
```suggestion
          cache_key: ${{ github.sha }}-${{ github.run_id }}-rpm-${{ matrix.distrib }}
```

File: .github/workflows/awie.yml [hunk 4]
```diff
@@ -60,21 +60,56 @@ jobs:
...
+          rpm_gpg_key: ${{ secrets.RPM_GPG_SIGNING_KEY }}
+          rpm_gpg_signing_key_id: ${{ secrets.RPM_GPG_SIGNING_KEY_ID }}
+          rpm_gpg_signing_passphrase: ${{ secrets.RPM_GPG_SIGNING_PASSPHRASE }}
+
+      - name: Upload package artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: packages-${{ matrix.distrib }}
+          path: ./*.${{ matrix.package_extension}}
+          retention-days: 1
```
Comment (kduret) [inline .github/workflows/awie.yml:112]: This was only needed for testing purpose
```suggestion
```

File: centreon-awie/packaging/centreon-awie.spectemplate [hunk 5]
```diff
@@ -1,69 +0,0 @@
...
-%{__install} -d -m 0770 %buildroot%{_localstatedir}/cache/centreon/filesUpload
-
-%clean
-rm -rf $RPM_BUILD_ROOT
-
-%files
-%defattr(-,root,root,-)
-%attr(-,apache,apache) %{centreon_www}/modules/%{name}
-%defattr(0664, centreon, centreon, 0770)
-%{_localstatedir}/cache/centreon/filesUpload
```
Comment (kduret) [inline centreon-awie/packaging/centreon-awie.spectemplate:63]: this directory is not created in nfpm package

File: centreon-awie/packaging/centreon-awie.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  Move Centreon Awie packaging to nfpm
+  Commit: @COMMIT_HASH@
+vendor: "Centreon"
+homepage: "https://www.centreon.com"
+license: "Apache-2.0"
+
+contents:
+  - src: "../www/modules/centreon-awie/*"
+    dst: "/usr/share/centreon/www/modules/centreon-awie"
+    file_info:
```
Comment (kduret) [inline centreon-awie/packaging/centreon-awie.yaml:20]: owner and group must be `apache` for almalinux and `www-data` for debian
currently, nfpm does not support env variable for owner and group, so you should replace it manually like it is done for commit hash : https://github.com/centreon/centreon/blob/develop/.github/actions/package-nfpm/action.yml#L74

File: centreon-awie/packaging/centreon-awie.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      owner: "@GROUP_OWNER@"
+      group: "@GROUP_OWNER@"
+
+  - dst: /var/cache/centreon/filesUpload
+    type: dir
+    file_info:
```
Comment (kduret) [inline centreon-awie/packaging/centreon-awie.yaml:26]: owner and group should be `centreon`

File: centreon-awie/packaging/centreon-awie.yaml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+vendor: "Centreon"
+homepage: "https://www.centreon.com"
+license: "Apache-2.0"
+
+contents:
+  - src: "../www/modules/centreon-awie/*"
+    dst: "/usr/share/centreon/www/modules/centreon-awie"
+    file_info:
+      owner: "@GROUP_OWNER@"
+      group: "@GROUP_OWNER@"
```
Comment (kduret) [inline centreon-awie/packaging/centreon-awie.yaml:22]:
```suggestion
      owner: "@APACHE_OWNER@"
      group: "@APACHE_OWNER@"
```

File: .github/workflows/awie.yml [hunk 9]
```diff
@@ -60,21 +60,51 @@ jobs:
...
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Set group and owner according to distrib
+        run: |
+          if [ "${{ matrix.distrib }}" = "bullseye" ]; then
+            sed -i "s/@GROUP_OWNER@/www-data/g" ./centreon-awie/packaging/centreon-awie.yaml
+          else
+            sed -i "s/@GROUP_OWNER@/apache/g" ./centreon-awie/packaging/centreon-awie.yaml
```
Comment (kduret) [inline .github/workflows/awie.yml:82]:
```suggestion
            sed -i "s/@APACHE_OWNER@/www-data/g" ./centreon-awie/packaging/centreon-awie.yaml
          else
            sed -i "s/@APACHE_OWNER@/apache/g" ./centreon-awie/packaging/centreon-awie.yaml
```

**Review by tuntoja — APPROVED** [review]

**Review by paul-oureib — APPROVED** [review]

**Pull request merged**